### ISSUE TITLE
Error setting up integration

### ISSUE THREAD

**Issue by jsermer** [legrego/homeassistant-elasticsearch#161]
**Environment**
Home-Assistant version: 2022.7.3
Elasticsearch version: 8.3.2

Relevant `configuration.yml` settings:
setup via ui

**Describe the bug**
```This error originated from a custom integration.

Logger: homeassistant.config_entries
Source: custom_components/elasticsearch/es_version.py:21
Integration: Elasticsearch (documentation, issues)
First occurred: 9:09:39 AM (1 occurrences)
Last logged: 9:09:39 AM

Error setting up entry https://10.43.16.30:9200 for elasticsearch
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 353, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/elasticsearch/__init__.py", line 147, in async_setup_entry
    init = await _async_init_integration(hass, config_entry)
  File "/config/custom_components/elasticsearch/__init__.py", line 177, in _async_init_integration
    await integration.async_init()
  File "/config/custom_components/elasticsearch/es_integration.py", line 29, in async_init
    await self.gateway.async_init()
  File "/config/custom_components/elasticsearch/es_gateway.py", line 91, in async_init
    await self.es_version.async_init()
  File "/config/custom_components/elasticsearch/es_version.py", line 21, in async_init
    self.build_flavor = version["build_flavor"]
KeyError: 'build_flavor'
```

**To Reproduce**

**Expected behavior**
Events still seem to be published but the queue size and health sensors show as 'unavailable'

**Screenshots**

**Additional context**

**Comment by legrego** [comment]
Thanks for the report, @jsermer. I can also reproduce this on version `8.3.2`. This is a regression on the Elasticsearch side, which should be resolved in `8.3.3` once it's released:
- https://github.com/elastic/elasticsearch/issues/88318
- https://github.com/elastic/elasticsearch/pull/88336

**Comment by legrego** [comment]
**update**: It appears that we no longer care about the `build_flavor` in this component. I've raised https://github.com/legrego/homeassistant-elasticsearch/pull/162 to attempt to resolve this on our side.
